doc/examples/list_obsoletes_plugin.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (C) 2015  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+
+"""A plugin that lists available packages filtered by their relation to the system."""
+
+import dnf
+import dnf.cli
+
+
+# The parent class allows registration to the CLI manager.
+class Command(dnf.cli.Command):
+
+    """A command that lists packages installed on the system that are
+       obsoleted by packages in any known repository."""
+
+    # An alias is needed to invoke the command from command line.
+    aliases = ['foo']  # <-- SET YOUR ALIAS HERE.
+
+    def configure(self, args):
+        """Setup the demands."""
+        # Repositories serve as sources of information about packages.
+        self.cli.demands.available_repos = True
+        # A sack is needed for querying.
+        self.cli.demands.sack_activation = True
+
+    def run(self, args):
+        """Run the command."""
+
+        obs_tuples = []
+        # A query matches all available packages
+        q = self.base.sack.query()
+
+        if not args:
+            # Narrow down query to match only installed packages
+            inst = q.installed()
+            # A query matches all packages that obsolete any other package
+            obsoletes = q.filter(obsoletes=inst)
+            for new in obsoletes:
+                obsoleted_reldeps = new.obsoletes
+                obs_tuples.extend([(new, old) for old in
+                                  inst.filter(provides=obsoleted_reldeps)])
```

Comment:
We need to document that provides accepts *xxx* (where *xxx* is the type returned by ```Package.obsoletes```) before merging.